arch/arc/core/fault_s.S

```diff
@@ -95,7 +95,11 @@ _exc_entry:
 	_create_irq_stack_frame
 
 #ifdef CONFIG_ARC_HAS_SECURE
+#ifdef CONFIG_ARC_SECURE_FIRMWARE
 	lr r0,[_ARC_V2_ERSEC_STAT]
+#else
+	mov r0, 0
```

Comment:
I think it's actually ok to read ERSEC_STAT from Normal mode as well - fields without access will be 0 (IOW/RAZ)

Reply:
read is ok, but no write. It will make codes simpler